File: BuckeyeMarketplace.API/Models/Order.cs
namespace BuckeyeMarketplace.API.Models;

public class Order
{
    public int Id { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string ConfirmationNumber { get; set; } = string.Empty;
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public decimal TotalAmount { get; set; }
    public string Status { get; set; } = "Placed";

    // Shipping info
    public string ShippingName { get; set; } = string.Empty;
    public string ShippingAddress { get; set; } = string.Empty;
    public string ShippingCity { get; set; } = string.Empty;
    public string ShippingState { get; set; } = string.Empty;
    public string ShippingZip { get; set; } = string.Empty;

    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: client/src/types/Order.ts
export interface OrderItem {
  id: number;
  orderId: number;
  productId: number | null;
  productTitle: string;
  unitPrice: number;
  quantity: number;
}

export interface Order {
  id: number;
  userId: string;
  confirmationNumber: string;
  orderDate: string;
  totalAmount: number;
  status: string;
  shippingName: string;
  shippingAddress: string;
  shippingCity: string;
  shippingState: string;
  shippingZip: string;
  orderItems: OrderItem[];
}

export interface OrderConfirmation {
  orderId: number;
  confirmationNumber: string;
  orderDate: string;
  totalAmount: number;
}

export interface CheckoutFormData {
  shippingName: string;
  shippingAddress: string;
  shippingCity: string;
  shippingState: string;
  shippingZip: string;
}
